vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -1,1011 +1,508 @@
-
-                mor = {
-                    "mor_type": mor_type,
-                    "mor": obj,
-                    "hostname": hostname,
-                    "tags": [t for t in tags + instance_tags if t.split(":", 1)[0] not in self.excluded_host_tags],
-                }
-                if self.excluded_host_tags:
-                    mor["excluded_host_tags"] = [
-                        t for t in tags + instance_tags if t.split(":", 1)[0] in self.excluded_host_tags
-                    ]
-                obj_list[vimtype].append(mor)
-
-        self.log.debug("All objects with attributes cached in %s seconds.", time.time() - start)
-        return obj_list
-
-    @staticmethod
-    def _is_excluded(obj, properties, regexes, include_only_marked):
-        """
-        Return `True` if the given host or virtual machine is excluded by the user configuration,
-        i.e. violates any of the following rules:
-        * Do not match the corresponding `*_include_only` regular expressions
-        * Is "non-labeled" while `include_only_marked` is enabled (virtual machine only)
-        """
-        # Host
-        if isinstance(obj, vim.HostSystem):
-            # Based on `host_include_only_regex`
-            if regexes and regexes.get('host_include') is not None:
-                match = re.search(regexes['host_include'], properties.get("name", ""), re.IGNORECASE)
-                if not match:
-                    return True
-
-        # VirtualMachine
-        elif isinstance(obj, vim.VirtualMachine):
-            # Based on `vm_include_only_regex`
-            if regexes and regexes.get('vm_include') is not None:
-                match = re.search(regexes['vm_include'], properties.get("name", ""), re.IGNORECASE)
-                if not match:
-                    return True
-
-            # Based on `include_only_marked`
-            if include_only_marked:
-                monitored = False
-                for field in properties.get("customValue", ""):
-                    if field.value == VM_MONITORING_FLAG:
-                        monitored = True
-                        break  # we shall monitor
-                if not monitored:
-                    return True
-
-        return False
-
-    def _cache_morlist_raw(self, instance):
-        """
-        Fill the Mor objects queue that will be asynchronously processed later.
-        Resolve the vCenter `rootFolder` and initiate hosts and virtual machines
-        discovery.
-        """
-        i_key = self._instance_key(instance)
-        self.log.debug("Caching the morlist for vcenter instance %s", i_key)
+    def initiate_connection(self):
+        try:
+            self.log.info("Connecting to the vCenter API...")
+            self.api = VSphereAPI(self.instance)
+            self.log.info("Connected")
+        except APIConnectionError:
+            self.log.exception("Cannot authenticate to vCenter API. The check will not run.")
```

Comment:
Since we already raised

```suggestion
            self.log.error("Cannot authenticate to vCenter API. The check will not run.")
```